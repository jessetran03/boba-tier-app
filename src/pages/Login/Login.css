.login label {
  margin-right: 10px;
}

.login input {
  margin: 2px;
  padding: 4px;
  border-radius: 15px;
  width: 200px;
}

.login button {
  margin: 20px 5px;
  font-size: 15px;
  font-weight: 500;
  border-radius: 15px;
  padding: 6px;
}

.login button:hover {
  cursor: pointer;
  background-color: #CCCCCC;
}