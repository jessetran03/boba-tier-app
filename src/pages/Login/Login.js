import React, { Component } from 'react'
import { Link } from 'react-router-dom'
//import TokenService from '../services/token-service'
import './Login.css'

export default class Login extends Component {
  state = {
    error: null
  }

  render() {
    const { error } = this.state
    return (
      <section className='login'>
        <form onSubmit={this.handleLogin} className='sign-form'>
          <h2>Sign in</h2>
          <div>
            {error && <p className='form-error'>{error}</p>}
          </div>
          <label htmlFor='username-input'>
            Username:
          </label>
          <input type='text' id='username-input' name='username' />
          <br />
          <label htmlFor='password-input' className='password-label'>
            Password:
          </label>
          <input type='password' id='password-input' name='password' />
          <br />
          <button
            type='submit'
          >
            Log in
          </button>
          <Link to='/register'>
            <button className='sign-up-button'>Sign Up</button>
          </Link>
        </form>
      </section>
    )
  }
}