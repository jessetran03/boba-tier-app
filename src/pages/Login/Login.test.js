import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Login from './Login'
import renderer from 'react-test-renderer'

it('renders Login without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <Login />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render Login without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <Login />
    </BrowserRouter>
  )
    .toJSON();
  expect(wrapper).toMatchSnapshot()
})