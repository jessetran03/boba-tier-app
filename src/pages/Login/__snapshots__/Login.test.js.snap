// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Login without data 1`] = `
<section
  className="login"
>
  <form
    className="sign-form"
    onSubmit={[Function]}
  >
    <h2>
      Sign in
    </h2>
    <div />
    <label
      htmlFor="username-input"
    >
      Username:
    </label>
    <input
      autoFocus={true}
      className="username-input"
      id="username-input"
      name="username"
      type="text"
    />
    <br />
    <label
      className="password-label"
      htmlFor="password-input"
    >
      Password:
    </label>
    <input
      className="password-input"
      id="password-input"
      name="password"
      type="password"
    />
    <br />
    <button
      type="submit"
    >
      Log in
    </button>
    <a
      href="/register"
      onClick={[Function]}
    >
      <button
        className="sign-up-button"
      >
        Sign up
      </button>
    </a>
  </form>
  <div />
</section>
`;
